### PR TITLE
[operator-trivy] update report-updater reference dictionary on a schedule

### LINKED ISSUE #5720: [operator-trivy] Update BDU every 4-6 hours
### Preflight Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/deckhouse/deckhouse/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://github.com/deckhouse/deckhouse/issues) for an issue that matches the one I want to file, without success.

### Use case. Why is this important?

...

### Proposed Solution

We should update BDU every 4-6 hours instead of update on pod start.

### Additional Information

### PR DESCRIPTION
**Pull request by miklezzzz**
## Description
Trivy-db-schedule github workflow has been extended to perform periodic updates and uploads of `CVE to BDU` dictionary to our registries.
Report-updater has been updated to support periodic dictionary updates (once per 6 hours by default). The update interval is a configurable parameter. Its deployment has been updated to have all necessary authentication required for getting `CVE to BDU` dictionary image from a repository.

## Why do we need it, and what problem does it solve?
Closes #5720
It allows us to have actual `CVE to BDU` dictionary for `report-updater` in a cluster.

## What is the expected result?
1) `CVE to BDU` dictionary is updated and pushed to our registries by the github action (every 6 hours by default)
2) report-updater keeps its dictionary updated by means of periodic updates (every 6 hours by default)

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: operator-trivy
type: chore
summary: Adds periodic updates of report-updater's dictionary.
impact_level: report-updater will be restarted.
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @miklezzzz. I've started the workflow for label 'e2e/run/yandex-cloud'...

File: ee/modules/500-operator-trivy/images/report-updater/web/hook/hook.go [hunk 1]
```diff
@@ -32,20 +33,27 @@ type patchOperation struct {
...
+		logger:     logger,
+		dictionary: dict,
 	}, nil
 }
 
-func (h *Handler) StartRenewBduCache(ch chan struct{}) error {
-	ch <- struct{}{}
-	return nil
+func (h *Handler) StartRenewCacheLoop() {
+	ticker := time.NewTicker(6 * time.Hour)
```
Comment (yalosev) [inline ee/modules/500-operator-trivy/images/report-updater/web/hook/hook.go:44]: Could we pass this as argument with default value (6 hours) please

File: ee/modules/500-operator-trivy/images/report-updater/cache/cache.go [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+		return nil, fmt.Errorf("failed to open docker config.json: %w", err)
+	}
+	defer dockerConfigFile.Close()
+
+	b, err := io.ReadAll(dockerConfigFile)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read docker config.json: %w", err)
+	}
+
+	err = json.Unmarshal(b, &dockerConfig)
```
Comment (yalosev) [inline ee/modules/500-operator-trivy/images/report-updater/cache/cache.go:97]:
```suggestion
	err = json.NewDecoder(dockerConfigFile).Decode(&dockerConfig)
```

File: ee/modules/500-operator-trivy/images/report-updater/cache/cache.go [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+
+	//iterate over descriptors to get the ones with relevant MediaType
+	for _, descriptor := range successors {
+		switch descriptor.MediaType {
+		case tarGzMediaType:
+			readCloser, err := store.Fetch(ctx, descriptor)
+			if err != nil {
+				fmt.Errorf("renewing BDU failed: couldn't fetch tar archive: %w", err)
+			}
+			defer readCloser.Close()
```
Comment (yalosev) [inline ee/modules/500-operator-trivy/images/report-updater/cache/cache.go:194]: `defer` in `for loop` is not safe, would be better to wrap it with a function. At least anonymous one

File: ee/modules/500-operator-trivy/images/report-updater/cache/cache.go [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+			b, err := io.ReadAll(tarReader)
+			if err != nil {
+				return fmt.Errorf("renewing BDU failed: couldn't read from tar: %w", err)
+			}
+
+			tempDict := &VulnerabilityDictionary{
+				Data: make(map[string][]string),
+			}
+
+			err = json.Unmarshal(b, &tempDict)
```
Comment (yalosev) [inline ee/modules/500-operator-trivy/images/report-updater/cache/cache.go:237]:
```suggestion
			err = json.NewDecoder(tarReader).Decode(&tempDict)
```

**Review by yalosev — CHANGES_REQUESTED** [review]
A few details

**Comment by miklezzzz** [comment]
nice! I've got the point, I guess..

**Review by name212 — APPROVED** [review]

**Pull request merged**